cl/corpus_importer/management/commands/make_aws_manifest_files.py

```diff
@@ -187,6 +210,341 @@ def get_custom_query(
+    )
+
+    # Extract the values of the specified field from the filtered queryset.
+    return list(queryset.values_list(field_name))
+
+
+def get_monthly_record_ids_by_type(
+    record_type: str, timestamp: datetime
+) -> list[tuple[int]]:
+    """
+    Retrieves a list of unique IDs for records of a specific type that were
+    created or modified after a given timestamp.
+
+    The function handles different record types by querying the main model
+    and any relevant nested/related models that might have been updated.
+
+    Args:
+        record_type: A string identifying the type of records to retrieve.
+        timestamp: A datetime object representing the month for which to
+            retrieve records.
+        options: A dictionary containing any additional options.
+
+    Returns:
+        A list of unique integer IDs for the records of the specified type
+        that were created or modified on or after the given `timestamp`.
+    """
+    related_field = "pk"
+    main_model = None
+    nested_models: list[type[AbstractDateTimeModel]] = []
+    match record_type:
+        case SEARCH_TYPES.OPINION:
+            main_model = Opinion
+            nested_models = []
+        case SEARCH_TYPES.ORAL_ARGUMENT:
+            main_model = Audio
+            nested_models = []
+        case SEARCH_TYPES.PEOPLE:
+            main_model = Person
+            nested_models = [
+                Source,
+                ABARating,
+                Education,
+                PoliticalAffiliation,
+                Position,
+            ]
+            related_field = "person_id"
+        case "fd":
+            main_model = FinancialDisclosure
+            nested_models = [
+                Agreement,
+                Debt,
+                Gift,
+                Investment,
+                NonInvestmentIncome,
+                Reimbursement,
+                SpouseIncome,
+                FDPosition,
+            ]
+            related_field = "financial_disclosure_id"
+        case _:
+            raise NotImplementedError(
+                f"Record type '{record_type}' is not supported."
+            )
+
+    record_ids = get_records_modified_since(main_model, "pk", timestamp)
+    # check nested/related models
+    for model in nested_models:
+        record_ids.extend(
+            get_records_modified_since(model, related_field, timestamp)
+        )
+
+    match record_type:
+        case SEARCH_TYPES.OPINION:
+            # Apply filters to include only opinions that are either:
+            # 1. Not extracted by OCR.
+            # 2. Associated with a cluster whose source contains HARVARD_CASELAW
+            ids = [x[0] for x in record_ids]
+            return list(
+                Opinion.objects.filter(pk__in=ids)
+                .filter(
+                    Q(extracted_by_ocr=False)
+                    | Q(cluster__source__icontains=SOURCES.HARVARD_CASELAW)
+                )
```

Comment:
Just a quick question: `--monthly-export` isn’t compatible with `--all-records`, right? So monthly exports always apply those filters, even if `--all-records` is provided, correct?

Reply:
> --monthly-export isn’t compatible with --all-records, right? So monthly exports always apply those filters, even if --all-records is provided, correct?

That's a good question. When I first wrote this method, I was mainly thinking about our current datasets, where some already have filters applied. I didn't think about the `--all-records` option. I'll add some code to make sure this helper function considers that flag included in the options dictionary.